packages/graphql/src/schema/validation/custom-rules/directives/relationship.ts

```diff
@@ -16,181 +16,136 @@
+                }
+            });
+        },
+        FieldDefinition(fieldDefinitionNode: FieldDefinitionNode, _key, _parent, path, ancestors) {
+            const appliedRelationship = fieldDefinitionNode.directives?.find(
+                (directive) => directive.name.value === relationshipDirective.name
             );
-
-            if (relationshipPropertiesType.length > 1) {
-                throw new DocumentValidationError(
-                    `@relationship.properties invalid. Cannot have more than 1 type represent the relationship properties.`,
-                    ["properties"]
-                );
+            if (!appliedRelationship) {
+                return;
             }
-            if (!relationshipPropertiesType.length) {
-                throw new DocumentValidationError(
-                    `@relationship.properties invalid. Cannot find type to represent the relationship properties: ${propertiesValue}.`,
-                    ["properties"]
-                );
+            const isValidLocation = fieldIsInNodeType({ path, ancestors, typeMapWithExtensions });
+            const [pathToHere, _traversedDef, parentOfTraversedDef] = getPathToNode(path, ancestors);
+            const typeArg = appliedRelationship.arguments?.find((a) => a.name.value === "type");
+            const directionArg = appliedRelationship.arguments?.find((a) => a.name.value === "direction");
+            const propertiesArg = appliedRelationship.arguments?.find((a) => a.name.value === "properties");
+            if (!typeArg || !directionArg) {
+                // delegate to DirectiveArgumentOfCorrectType rule
+                return;
             }
-            const isRelationshipPropertiesTypeAnnotated = relationshipPropertiesType[0]?.directives?.some(
-                (d) => d.name.value === "relationshipProperties"
-            );
 
-            if (!isRelationshipPropertiesTypeAnnotated) {
-                throw new DocumentValidationError(
-                    `@relationship.properties invalid. Properties type ${propertiesValue} must use directive \`@relationshipProperties\`.`,
-                    ["properties"]
+            const { isValid, errorMsg, errorPath } = assertValid(() => {
+                if (!isValidLocation) {
+                    if (fieldIsInInterfaceType({ path, ancestors, typeMapWithExtensions })) {
+                        // throw more specific error for interface types as in the past it was possible to have relationships on interfaces
+                        throw new DocumentValidationError(
+                            `Invalid directive usage: Directive @${relationshipDirective.name} is not supported on fields of interface types (${parentOfTraversedDef?.name.value}). Since version 5.0.0, interface fields can only have @${declareRelationshipDirective.name}. Please add the @relationship directive to the fields in all types which implement it.`,
+                            []
+                        );
```

Comment:
Do we still want to keep the `Since version 5.0.0` message? It is a bit outdated by now

Reply:
maybe not, but I would change it separately from this PR to isolate the scope as much as possible!